File: ios/ResearchIntelApp/ResearchIntelApp/Models/Article.swift
import Foundation

struct Article: Identifiable, Codable, Hashable {
    let id: String
    let source: String
    let sourceId: String?
    let title: String
    let chineseTitle: String?
    let abstract: String?
    let summary: String?
    let url: String
    let doi: String?
    let publishedAt: Date?
    let topics: [String]
    let relevanceScore: Double
    let qualityScore: Double
    let isPreprint: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case source
        case sourceId = "source_id"
        case title
        case chineseTitle = "chinese_title"
        case abstract
        case summary
        case url
        case doi
        case publishedAt = "published_at"
        case topics
        case relevanceScore = "relevance_score"
        case qualityScore = "quality_score"
        case isPreprint = "is_preprint"
    }
}

File: backend/app/schemas/article.py
from datetime import datetime

from pydantic import BaseModel, Field, HttpUrl


class ArticleRead(BaseModel):
    id: str
    source: str
    source_id: str | None = None
    title: str
    chinese_title: str | None = None
    abstract: str | None = None
    summary: str | None = None
    url: HttpUrl | str
    doi: str | None = None
    published_at: datetime | None = None
    topics: list[str] = Field(default_factory=list)
    relevance_score: float = 0.0
    quality_score: float = 0.0
    is_preprint: bool = False
